### PR TITLE
K8s Net pol -- UDP handling

### PR DESCRIPTION
**Pull request by seswarrajan**
Signed-off-by: Eswar Rajan Subramanian <eswar@accuknox.com>

### CODE REVIEW

File: src/networkpolicy/networkPolicy.go [hunk 1]
```diff
@@ -1598,13 +1598,31 @@ func mergeIngressPolicies(existPolicy types.KnoxNetworkPolicy, policies []types.
 						}
 					}
 				}
+			} else if len(newIngress.FromCIDRs) > 0 && len(newIngress.ToPorts) > 0 {
```
Comment (nyrahul) [inline src/networkpolicy/networkPolicy.go:1633]: In this condition only `FromCIDRs` at index 0 is checked. What happens if we get multiple `FromCIDRs` in the rule.

File: src/networkpolicy/networkPolicy.go [hunk 2]
```diff
@@ -1668,7 +1686,24 @@ func mergeEgressPolicies(existPolicy types.KnoxNetworkPolicy, policies []types.K
 						}
 					}
 				}
+			} else if len(newEgress.ToCIDRs) > 0 && len(newEgress.ToPorts) > 0 {
```
Comment (nyrahul) [inline src/networkpolicy/networkPolicy.go:1708]: This is duplicate code. Please keep an API for CIDR matching and call from both the ToCIDRs and FromCIDRs condition.

File: src/networkpolicy/networkPolicy.go [hunk 3]
```diff
@@ -1858,6 +1893,49 @@ func convertKnoxNetworkLogToKnoxNetworkPolicy(log *types.KnoxNetworkLog, pods []
...
+
+		egress.ToPorts = append(egress.ToPorts, types.SpecPort{
+			Port:     strconv.Itoa(log.DstPort),
+			Protocol: libs.GetProtocol(log.Protocol),
+		})
+
+		ePolicy.Spec.Egress = append(ePolicy.Spec.Egress, egress)
+		ePolicy.Metadata["namespace"] = log.SrcNamespace
+		egressPolicy = &ePolicy
+	} else if log.DstPodName == "" && len(log.DstReservedLabels) == 0 && log.Direction == "INGRESS" {
```
Comment (nyrahul) [inline src/networkpolicy/networkPolicy.go]: There is too much of duplicate code in the two condition. Can we create a function to handle this?

File: src/plugin/k8sNetwork.go [hunk 4]
```diff
@@ -31,32 +28,52 @@ func ConvertKnoxNetPolicyToK8sNetworkPolicy(clustername, namespace string) []nv1
...
+				} else {
+					port = nv1.NetworkPolicyPort{
+						Protocol: &protocol,
+					}
+				}
+
+				if len(eg.ToCIDRs) > 0 {
+					to = nv1.NetworkPolicyPeer{
+						IPBlock: &nv1.IPBlock{
+							CIDR: eg.ToCIDRs[0].CIDRs[0],
```
Comment (nyrahul) [inline src/plugin/k8sNetwork.go]: What happens if `len(eg.ToCIDRs) > 1`?

File: src/plugin/k8sNetwork.go [hunk 5]
```diff
@@ -71,27 +88,46 @@ func ConvertKnoxNetPolicyToK8sNetworkPolicy(clustername, namespace string) []nv1
...
+						},
+						Protocol: &protocol,
+					}
+				} else {
+					port = nv1.NetworkPolicyPort{
+						Protocol: &protocol,
+					}
+				}
+
+				if len(ing.FromCIDRs) > 0 {
```
Comment (nyrahul) [inline src/plugin/k8sNetwork.go]: Again there is duplication of code here from ToCIDR and FromCIDR perspective.

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: src/networkpolicy/networkPolicy.go (hunk 1 above)
Comment (seswarrajan) [reply src/networkpolicy/networkPolicy.go:1633]: Each egress rule contains only one CIDRs. Hence no issue.

**Review by nyrahul — APPROVED** [review]

**Pull request merged**